### PR TITLE
Forward Declare Enums

### PR DESCRIPTION
**Pull request by geoffthemedio**
Works for me, but want to check with others before merging...

### CODE REVIEW

**Comment by adrianbroher** [comment]
I don't see the point of this change. It doesn't really reduce circular inclusion dependencies.

**Comment by geoffthemedio** [comment]
It's not complete, so there are more removals of including Enums.h to be done. It does / will avoid having to include all the enum declarations into numerous headers (and thus source files) where they're not needed. I'm not concerned about circular include chains since the include guards should stop that for actually extending the resulting compiled source file.

**Comment by adrianbroher** [comment]
Let me rephrase:

I don't see the point of the PR as it doesn't significantly reduce the number of included files when replacing `universe/Enums.h` with `universe/EnumsFwd.h`.  If you want to improve build time start with carving out the ValueRefBase, ConditionBase and EffectBase and place them into separate headers to avoid dependency inclusion pulled in by the concrete implementations.

**Comment by geoffthemedio** [comment]
It doesn't reduce the number of included files, but is substantially reduces the complexity and length of what is included.

There already is https://github.com/freeorion/freeorion/blob/master/universe/ValueRefFwd.h

**Comment by geoffthemedio** [comment]
Not sure how reproducible the Travis CI build times are, but at least right now this branch built a few min faster than the latest master.

**Comment by geoffthemedio** [comment]
I've also split ConditionBase into a separate header from the rest of the Condition classes, but it isn't actually very helpful, since Condition.h isn't included in very many places where only ConditionBase is needed.

**Comment by adrianbroher** [comment]
> I don't see the point of the PR as it doesn't significantly reduce the number of included files when replacing universe/Enums.h with universe/EnumsFwd.h. 

> It doesn't reduce the number of included files, but is substantially reduces the complexity and length of what is included.

I stand corrected then.

> Not sure how reproducible the Travis CI build times are, but at least right now this branch built a few min faster than the latest master.

Good job Geoff.
